index.bs

```diff
@@ -697,53 +702,68 @@ When this method is invoked, the user agent MUST execute the following algorithm
-                :   {{AuthenticatorResponse/clientDataJSON}}
-                ::  A new {{ArrayBuffer}}, created using |global|'s [=%ArrayBuffer%=], containing the bytes of |clientDataJSON|.
-                :   {{AuthenticatorAttestationResponse/attestationObject}}
-                ::  |attestationObject|
-            :   {{PublicKeyCredential/clientExtensionResults}}
-            ::  A new {{AuthenticationExtensions}} object containing the [=extension identifier=] → [=client extension output=]
-                entries created by running each extension's [=client extension processing=] algorithm to create the [=client
-                extension outputs=], for each [=client extension=] in {{AuthenticatorResponse/clientDataJSON}}.clientExtensions.
+    :   If any |authenticator| indicates success,
+    ::  1.  [=set/Remove=] |authenticator| from |issuedRequests|.
+
+        2.  Let |credentialCreationData| be a [=struct=] whose [=items=] are:
+
+            :   <code><dfn>attestationObjectResult</dfn></code>
+            ::  whose value is the bytes returned from the successful [=authenticatorMakeCredential=] operation.
+
+                Note: this value is <code>attObj</code>, as defined in [[#generating-an-attestation-object]].
+
+            :   <code><dfn>clientDataJSONResult</dfn></code>
+            ::  whose value is the bytes of |clientDataJSON|.
+
+            :   <code><dfn>extensionOutputsMap</dfn></code>
+            ::  whose value is an [=ordered map=] with [=map/keys=] of type [=extension identifier=]
+                and [=map/values=] of type [=client extension output=].
+                <code>[=extensionOutputsMap=]</code>'s [=map/entries=] are
+                created by running each extension's [=client extension processing=] algorithm to create the [=client
+                extension outputs=], for each [=client extension=] in <code>{{AuthenticatorResponse/clientDataJSON}}.clientExtensions</code>.
+
+        3.  Let |value| be a [=struct=] whose [=items=] are:
+
+            :   <code><dfn>constructResultantCredentialCallback</dfn></code>
+            ::  whose value is a Web IDL {{Function}} [=callback function type=] value implementing the
+                steps defined in [[#constructResultantCredentialCallback]].
+
+            :   <code><dfn>credentialCreationDataInstance</dfn></code>
+            ::  whose value is |credentialCreationData|.
+
+            ISSUE: <code>[=constructResultantCredentialCallback=]</code> and
+                <code>[=credentialCreationDataInstance=]</code> should actually not be defined here (?) and link back
+                to their definitions in [[!CREDENTIAL-MANAGEMENT-1]] (?).
 
         5. [=set/For each=] remaining |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=] operation on
             |authenticator| and [=set/remove=] it from |issuedRequests|.
+
         6. Return |value| and terminate this algorithm.
```

Comment:
You can't really "return" from in parallel steps.

Reply:
I see. https://wicg.github.io/background-fetch/#attempt-a-background-fetch-algorithm is really educational and helps quite a bit. But I prefer not to correct this in the abort PR. We can open up another PR to do parallel right. 

@mikewest @equalsJeffH does this mean we need to correct both credman and webauthn?

Reply:
> does this mean we need to correct both credman and webauthn?

I'm not sure.  This algorithm is invoked from credman as an internal method. The associated mods to credman are queued as credman PR https://github.com/w3c/webappsec-credential-management/pull/100  The credman steps that invoke this webauthn method/algorithm are themselves running in parallel. So saying "return" here seemed fine.